tests/test_qc_metrics.py

```diff
@@ -188,47 +188,45 @@ def test_dask_against_in_memory(adata, log1p):
     assert_equal(adata, adata_as_dask)
 
 
-def adata_mito():
-    a = np.random.binomial(100, 0.005, (1000, 1000))
-    init_var = pd.DataFrame(
-        dict(mito=np.concatenate((np.ones(100, dtype=bool), np.zeros(900, dtype=bool))))
+@pytest.fixture
+def adata_mito() -> AnnData:
+    return AnnData(
+        X=np.random.binomial(100, 0.005, (1000, 1000)),
+        var=dict(
+            mito=np.concatenate((np.ones(100, dtype=bool), np.zeros(900, dtype=bool)))
+        ),
     )
-    adata_dense = AnnData(X=a, var=init_var.copy())
-    return adata_dense, init_var
 
 
 @pytest.mark.parametrize("cls", ARRAY_TYPES_MEM)
-def test_qc_metrics_format(cls):
-    adata_dense, init_var = adata_mito()
-    sc.pp.calculate_qc_metrics(adata_dense, qc_vars=["mito"], inplace=True)
-    adata = AnnData(X=cls(adata_dense.X), var=init_var.copy())
+def test_qc_metrics_format(cls, adata_mito: AnnData) -> None:
+    sc.pp.calculate_qc_metrics(adata_mito, qc_vars=["mito"], inplace=True)
+    adata = AnnData(X=cls(adata_mito.X), var=adata_mito.var.copy())
```

Comment:
```suggestion
    var_clean = adata_mito.var.copy()
    sc.pp.calculate_qc_metrics(adata_mito, qc_vars=["mito"], inplace=True)
    adata = AnnData(X=cls(adata_mito.X), var=var_clean)
```
I assume the point of the test was to check if the calculated qc metrics for `var` are identical if you give it the same `var` twice, but with these changes we are giving the a copy of what was just run through `calculate_qc_metrics`

Reply:
good point, we should use the .var before running this.